src/lib/Tests/Pagination/Pagerfanta/TrashItemAdapterTest.php

```diff
@@ -0,0 +1,143 @@
+            ->getMockBuilder(SortClause::class)
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        // Count query will necessarily have a 0 limit.
+        $countQuery = clone $query;
+        $countQuery->limit = 0;
+
+        $searchResult = new SearchResult(['count' => $nbResults]);
+        $this->trashService
+            ->expects($this->once())
+            ->method('findTrashItems')
+            ->with($this->equalTo($countQuery))
+            ->will($this->returnValue($searchResult));
+
+        $adapter = $this->getAdapter($query, $this->trashService);
+        $this->assertSame($nbResults, $adapter->getNbResults());
+        // Running a 2nd time to ensure SearchService::findContent() is called only once.
+        $this->assertSame($nbResults, $adapter->getNbResults());
+    }
+
+    public function testGetSlice()
+    {
+        $offset = 20;
+        $limit = 25;
+        $nbResults = 123;
+
+        $query = new Query();
+        $query->query = $this->createMock(CriterionInterface::class);
+        $query->sortClauses = $this
+            ->getMockBuilder(SortClause::class)
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        // Injected query is being cloned to modify offset/limit,
+        // so we need to do the same here for our assertions.
+        $searchQuery = clone $query;
+        $searchQuery->offset = $offset;
+        $searchQuery->limit = $limit;
+        $searchQuery->performCount = false;
+
+        $items = [];
+        for ($i = 0; $i < $limit; ++$i) {
+            $content = $this->getMockForAbstractClass(APIContent::class);
+            $items[] = $content;
+        }
+        $finalResult = $this->getExpectedFinalResultFromHits($items);
+        $searchResult = new SearchResult(['items' => $items, 'count' => $nbResults]);
+        $this
+            ->trashService
+            ->expects($this->once())
+            ->method('findTrashItems')
+            ->with($this->equalTo($searchQuery))
+            ->will($this->returnValue($searchResult));
+
+        $adapter = $this->getAdapter($query, $this->trashService);
+        $this->assertSame($finalResult, $adapter->getSlice($offset, $limit));
+        $this->assertSame($nbResults, $adapter->getNbResults());
+        // Running a 2nd time to ensure SearchService::findContent() is called only once.
+        $this->assertSame($nbResults, $adapter->getNbResults());
+    }
+
+    public function testGetNbResultsWhenSet()
+    {
+        $nbResults = 123;
+
+        $query = new Query();
+
+        $this->trashService
+            ->expects($this->never())
+            ->method('findTrashItems');
+
+        $adapter = $this->getAdapter($query, $this->trashService);
+
+        $refObject = new \ReflectionObject($adapter);
+        $refProperty = $refObject->getProperty('nbResults');
+        $refProperty->setAccessible(true);
+        $refProperty->setValue($adapter, $nbResults);
```

Comment:
I believe if you actually have to use Reflections in your code you should consider doing a refactoring because in most cases this is a sign of badly written code. Given `TrashItemAdapter::$nbResults` is a private property, it should not be tested or altered in your tests.

Reply:
Yes, as @ViniTou point, this behavior is tested implicit in testGetNbResults method